turtle_tf2_cpp/src/turtle_tf2_listener.cpp

```diff
@@ -88,42 +67,80 @@ class FrameListener : public rclcpp::Node
-        toFrameRel, fromFrameRel,
-        tf2::TimePoint(),
-        500ms);
-    } catch (tf2::LookupException & ex) {
-      RCLCPP_INFO(this->get_logger(), "transform not ready");
+    if (!service_called_) {
+      // Check if the service is ready
+      if (!spawner_->service_is_ready()) {
+        return;
+      }
+
+      // Initialize request with turtle name and coordinates
+      // Note that x, y and theta are defined as floats in turtlesim/srv/Spawn
+      auto request = std::make_shared<turtlesim::srv::Spawn::Request>();
+      request->x = 4.0;
+      request->y = 2.0;
+      request->theta = 0.0;
+      request->name = "turtle2";
+      // Call request
+      result_ = spawner_->async_send_request(request);
+      service_called_ = true;
       return;
+    } else if (service_called_ && !turtle_spawned_) {
+      RCLCPP_INFO(this->get_logger(), "Successfully spawned %s", result_.get()->name.c_str());
+      turtle_spawned_ = true;
     }
 
-    geometry_msgs::msg::Twist msg;
+    if (!transform_available_) {
+      if (!tf_buffer_->canTransform(toFrameRel, fromFrameRel, tf2::TimePointZero, 0ms)) {
+        return;
+      }
+      RCLCPP_INFO(
+        get_logger(), "Transform %s->%s available",
+        toFrameRel.c_str(), fromFrameRel.c_str());
+      transform_available_ = true;
+    }
+
+    if (transform_available_) {
+      geometry_msgs::msg::TransformStamped transformStamped;
+
+      // Look up for the transformation between target_frame and turtle2 frames
+      // and send velocity commands for turtle2 to reach target_frame
+      try {
+        transformStamped = tf_buffer_->lookupTransform(
+          toFrameRel, fromFrameRel,
+          tf2::TimePointZero,
+          0ms);
+      } catch (tf2::LookupException &) {
+        RCLCPP_INFO(this->get_logger(), "Transform not ready");
+        return;
+      }
 
-    static const double scaleRotationRate = 1.0;
-    msg.angular.z = scaleRotationRate * atan2(
-      transformStamped.transform.translation.y,
-      transformStamped.transform.translation.x);
+      geometry_msgs::msg::Twist msg;
 
-    static const double scaleForwardSpeed = 0.5;
-    msg.linear.x = scaleForwardSpeed * sqrt(
-      pow(transformStamped.transform.translation.x, 2) +
-      pow(transformStamped.transform.translation.y, 2));
+      static const double scaleRotationRate = 1.0;
+      msg.angular.z = scaleRotationRate * atan2(
+        transformStamped.transform.translation.y,
+        transformStamped.transform.translation.x);
 
-    publisher_->publish(msg);
+      static const double scaleForwardSpeed = 0.5;
+      msg.linear.x = scaleForwardSpeed * sqrt(
+        pow(transformStamped.transform.translation.x, 2) +
+        pow(transformStamped.transform.translation.y, 2));
+
+      publisher_->publish(msg);
+    }
   }
-  rclcpp::TimerBase::SharedPtr timer_;
-  std::shared_ptr<tf2_ros::TransformListener> transform_listener_;
+  bool turtle_spawned_;
```

Comment:
con you add some doc to these variables?

Reply:
https://github.com/ros/geometry_tutorials/pull/53/commits/4dd474e6e6b00e520209b374cf7b31a0415739fc